src/logm.jl

```diff
@@ -0,0 +1,23 @@
+@inline log(A::StaticMatrix) = _log(Size(A), A)
+
+@inline function _log(::Size{(0,0)}, A::StaticMatrix)
+    T = typeof(log(one(eltype(A))))
+    SMT = similar_type(A,T)
+    return SMT()
+end
+
+@inline function _log(::Size{(1,1)}, A::StaticMatrix)
+    T = typeof(log(one(eltype(A))))
+    SMT = similar_type(A,T)
+    return SMT((log(A[1]), ))
+end
+
+function _log(::Size, A::StaticMatrix)
+    eigA = eigen(A)
+    T = typeof(log(one(eltype(A)) * one(eltype(eigA.values))))
+    VT = similar_type(typeof(eigA.values),T)
+    log_values = log.(VT(eigA.values))
+    log_eigA = Eigen(log_values, eigA.vectors)
+    logA = AbstractMatrix(log_eigA)
+    return logA
+end
```

Comment:
Why does this use a different algorithm than https://github.com/JuliaLang/julia/blob/d0b85902194fadaad78c7c41e95b5318e13c4062/stdlib/LinearAlgebra/src/dense.jl#L765-L791 ?

Reply:
The algorithm used there has two disadvantages:
- It is not type-stable; it checks whether the matrix is symmetric, and returns a real or complex results depending on the input matrix values
- It requires algorithms that are not implemented for `SMatrix`, or which fall back to converting to `Matrix` (Schur decomposition)

I want to use `log` for small matrices (e.g. 2x2), and type stability and not allocating memory is important to me.

The first could be fixed (by not checking the input matrix values). The second might be straightforward to fix by porting the Schur decomposition and its dependencies (Givens rotations) to StaticArrays, but this seemed like a larger project, and I don't know these implementations well enough to see whether they would work for `SMatrix` in a straightforward manner. Using an eigendecomposition instead seemed simpler.